runtime/testy/stateManager.py

```diff
@@ -91,21 +98,40 @@ def setValue(self, value, keys):
     currDict = self.state
     i = 0
     try:
+      path = []
       for key in enumerate(keys[:-1]):
         i = key[0]
-        currDict = currDict[key[1]]
+        path.append(currDict[key[1]])
+        currDict = currDict[key[1]][0]
       if len(keys) > 1:
         i += 1
       if keys[i] not in currDict:
         raise Exception
-      currDict[keys[i]] = value
+      path.append(currDict[keys[i]])
+      currDict[keys[i]][0] = value
+      currTime = time.time();
+      for item in path:
+        item[1] = currTime
       self.processMapping[PROCESS_NAMES.STUDENT_CODE].send(value)
     except:
       error = StudentAPIKeyError(self.dictErrorMessage(i, keys, currDict))
       self.processMapping[PROCESS_NAMES.STUDENT_CODE].send(error)
 
+  def getTimestamp(self, keys):
+    currDict = self.state
+    timestamp = 0
+    try:
+      for key in enumerate(keys):
```

Comment:
we can get rid of the first line following for loop with
`for i, key in enumerate(keys):`

You should apply this to the other places you have `i=key[0]` as well.